### PR TITLE
Update config files and other minor code polishing

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by FriederikeHanssen — APPROVED** [review]
Changelog

**Comment by FriederikeHanssen** [comment]
Don't see anything changed with closures though

**Comment by asp8200** [comment]
Do you intend to include this PR in Sarek 3.4.0?

**Comment by maxulysse** [comment]
> Do you intend to include this PR in Sarek 3.4.0?

Yes

File: conf/modules/markduplicates.config [hunk 1]
```diff
@@ -53,10 +53,10 @@ process {
         ext.prefix       = { "${meta.id}.sorted" }
         publishDir       = [
             // Never publish if BAM only should be published
-            enabled: !params.save_output_as_bam,
```
Comment (FriederikeHanssen) [inline conf/modules/markduplicates.config:56]: why remove the enabled?

File: conf/modules/modules.config [hunk 2]
```diff
@@ -40,6 +40,15 @@ process {
...
 
+    withName: 'MULTIQC' {
+        ext.args   = params.multiqc_title ? "--title \"$params.multiqc_title\"" : ''
+        publishDir = [
+            path: { "${params.outdir}/multiqc" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
```
Comment (FriederikeHanssen) [inline conf/modules/modules.config:51]: Where is this one comig from?

File: conf/modules/markduplicates.config (hunk 1 above)
Comment (maxulysse) [reply conf/modules/markduplicates.config:56]: not working in a closure, so it's now converted to a saveAs

**Review by FriederikeHanssen — COMMENTED** [review]
looks good, but why remove the enabled?

File: conf/modules/markduplicates.config (hunk 1 above)
Comment (FriederikeHanssen) [reply conf/modules/markduplicates.config:56]: ah ok

File: subworkflows/nf-core/bam_ngscheckmate/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    withName: ".*BAM_NGSCHECKMATE:BCFTOOLS_MPILEUP" {
+        ext.args2 = '--no-version --ploidy 1 -c'
+        ext.args3 = '--no-version'
+    }
+
+    withName: ".*BAM_NGSCHECKMATE:NGSCHECKMATE_NCM" {
+        ext.args = '-V'
+    }
+
+}
```
Comment (FriederikeHanssen) [inline subworkflows/nf-core/bam_ngscheckmate/nextflow.config:13]: 😬 this one is not loaded i hope

Comment (maxulysse) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config:13]: not at the moment

File: conf/modules/modules.config (hunk 2 above)
Comment (maxulysse) [reply conf/modules/modules.config:51]: TEMPLATE, somehow it vanished

**Comment by maxulysse** [comment]
> looks good, but why remove the enabled?

we can't have closure with `enabled`, so converted to `saveAs` with which we can use closure

File: subworkflows/nf-core/bam_ngscheckmate/nextflow.config (hunk 3 above)
Comment (maxulysse) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config]: Now we do

Comment (maxulysse) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config]: But it's the updated, so it's the one we used to use, nothing should change there

Comment (FriederikeHanssen) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config]: I don't like it. it gets to hard to find anything. In any case it then also needs closures

Comment (FriederikeHanssen) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config]: there is a ngscheckmate cnfig though in conf/ that has these values, so I don't think we need to include it

Comment (FriederikeHanssen) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config]: on dev anyways, and looks like that one needs closures too.

Comment (maxulysse) [reply subworkflows/nf-core/bam_ngscheckmate/nextflow.config:13]: Kept the old one as it was then

**Review by FriederikeHanssen — APPROVED** [review]

File: conf/modules/ngscheckmate.config [hunk 4]
```diff
@@ -1,26 +1,26 @@
+// IMPORTANT: Add this configuration to your modules.config
+
```
Comment (maxulysse) [inline conf/modules/ngscheckmate.config:2]:
```suggestion
```

**Review by FriederikeHanssen — APPROVED** [review]

**Review by FriederikeHanssen — APPROVED** [review]

File: conf/modules/msisensorpro.config [hunk 5]
```diff
@@ -16,10 +16,10 @@
 process {
 
     withName: 'MSISENSORPRO_MSISOMATIC' {
-        publishDir       = [
+        publishDir = [
             mode: params.publish_dir_mode,
             path: { "${params.outdir}/variant_calling/msisensorpro/${meta.id}/" },
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+                        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
```
Comment (maxulysse) [inline conf/modules/msisensorpro.config:22]:
```suggestion
                saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
```

File: conf/modules/msisensorpro.config [hunk 6]
```diff
@@ -16,10 +16,10 @@
 process {
 
     withName: 'MSISENSORPRO_MSISOMATIC' {
-        publishDir       = [
+        publishDir = [
             mode: params.publish_dir_mode,
             path: { "${params.outdir}/variant_calling/msisensorpro/${meta.id}/" },
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+                saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
```
Comment (maxulysse) [inline conf/modules/msisensorpro.config:22]:
```suggestion
            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
```

File: conf/modules/markduplicates.config [hunk 7]
```diff
@@ -91,42 +91,42 @@ process {
     }
 
     withName: 'GATK4SPARK_MARKDUPLICATES' {
-        ext.args         = '--remove-sequencing-duplicates false -VS LENIENT'
-        ext.prefix       = { "${meta.id}.md.cram" }
-        publishDir       = [
-            enabled: !params.save_output_as_bam,
+        ext.args   = '--remove-sequencing-duplicates false -VS LENIENT'
```
Comment (FriederikeHanssen) [inline conf/modules/markduplicates.config:94]: Missing curly brackets

Comment (maxulysse) [reply conf/modules/markduplicates.config:94]:
```suggestion
        ext.args   = { '--remove-sequencing-duplicates false -VS LENIENT' }
```

Comment (maxulysse) [reply conf/modules/markduplicates.config:94]: we dont' really need them in that case, but let's be consistent

**Review by SPPearce — APPROVED** [review]

**Pull request merged**